### PR TITLE
RDM-11001 - FT fixes for error response.

### PR DESCRIPTION
**Pull request by smathangi**
https://tools.hmcts.net/jira/browse/RDM-11001

This PR doesn't address correctness of functional tests. Just trying to baseline build (FTs) with current implementation. 
 
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/controller/advice/CaseDocumentControllerAdvice.java [hunk 1]
```diff
@@ -57,31 +59,52 @@
     @ExceptionHandler(RequiredFieldMissingException.class)
     protected ResponseEntity<Object> handleRequiredFieldMissingException(RequiredFieldMissingException exception) {
         return errorDetailsResponseEntity(exception, BAD_REQUEST,
-            ErrorConstants.INVALID_REQUEST.getErrorCode(), ErrorConstants.INVALID_REQUEST.getErrorMessage());
+            ErrorConstants.BAD_REQUEST.getErrorCode(), ErrorConstants.BAD_REQUEST.getErrorMessage());
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/documentam/controller/advice/CaseDocumentControllerAdvice.java:62]: i don't see a point of these custom errors like `INVALID_REQUEST` `RESOURCE_NOT_FOUND`.  It has already standard ones "BAD_REQUEST" and "NOT_FOUND"

**Comment by smathangi** [comment]
Some functional tests are still failing because of gap in implementation. (@S-001, @S-020, @S-023). 
Issue: caseId doesn't exist in the metadata returned from dm-store endpoint 
https://github.com/hmcts/ccd-case-document-am-api/blob/master/src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java#L145-L148

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java [hunk 2]
```diff
@@ -571,4 +571,8 @@ private void catchException(HttpClientErrorException exception) {
             throw new ServiceException(Constants.EXCEPTION_ERROR_MESSAGE, exception);
         }
     }
+
+    private String formateNotFoundMessage(String resourceId) {
```
Comment (sullivangareth) [inline src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java:575]: typo in method name ?

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java [hunk 3]
```diff
@@ -549,7 +549,7 @@ private HttpHeaders getHttpHeaders() {
 
     private void catchException(HttpClientErrorException exception, String messageParam) {
```
Comment (sullivangareth) [inline src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java:550]: rename to describe what it does better `catchAndRethrowException` ?

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java:550]: I agree with @sullivangareth that we could rename this method for clarity

**Review by tomxelliott — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java (hunk 3 above)
Comment (smathangi) [reply src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java:550]: sure, that's unchanged line 😄 . 
There are few other methods like that in that class.. will rename all together

**Pull request merged**